daml/Marketplace/Custody/Model.daml

```diff
@@ -145,3 +164,11 @@ createNewVersionOfDescription description postAssetVersion serializedRemainingCl
     claims = serializedRemainingClaims
  where
   adjustedId = description.assetId with version = postAssetVersion
+
+-- TODO this is still not unique e.g. if I have two asset deposits with the same asset
```

Comment:
I think this _is_ going to be a problem. We need to be consistent with any tradeId's we're creating in the rest of the program. I'm not sure how this is achieved. I'm guessing a state monad, although that might be a bottleneck. We need to check with team. Maybe @brian-weir knows?

Reply:
We can also use a singleton 'Counter Contract' (with a choice that gives you the next number: 1, 2, 3, ...) to generate unique identifiers. (`a state monad` - note that between two calls of the same choice, the state is not automatically persisted so we need something on the ledger, IMO)

As far as I understand lifecycling/processing is done by some central entity. It is also possible to add a 'unique identifier' parameter to the choice that needs to be specified by the central entity. So this way that entity can pick the method used to generate unique IDs (it can just choose the counter contractor use some internal unique ID, time etc..).

Reply:
Basically, with the latter paragraph I was saying push the problem onto the caller of Lifecycle/Process :dancers:

Reply:
> It is also possible to add a 'unique identifier' parameter to the choice that needs to be specified by the central entity. 

I think that's what @brian-weir just told me is the approach elsewhere in the marketplace

Reply:
@lucianojoublanc-da I added a unique payout ID (as a parameter) and (in tests) gave some examples of its usage.